File: apps/android/core/ai/src/main/java/com/scribblefit/core/ai/security/SecureKeyStorage.kt
package com.scribblefit.core.ai.security

interface SecureKeyStorage {
    suspend fun saveApiKey(key: String)
    suspend fun getApiKey(): String?
    suspend fun clearApiKey()
    
    suspend fun saveAuthToken(token: String)
    suspend fun getAuthToken(): String?
    suspend fun clearAuthToken()
}

File: apps/ios/ScribbleFit/ScribbleFit/Features/AI/Domain/SecureKeyStorage.swift
import Foundation

public protocol SecureKeyStorage: Sendable {
    func saveApiKey(_ key: String) async throws
    func getApiKey() async throws -> String?
    func clearApiKey() async throws
    
    func saveAuthToken(_ token: String) async throws
    func getAuthToken() async throws -> String?
    func clearAuthToken() async throws
}
